docs/DeveloperGuide.md

```diff
@@ -1,34 +1,247 @@
+
+![image_info](./pictures/FlowchartsetStatus.png)
+
+##### autoClean
+
+Step 1. Check activation status of StorageCleaner through the class level attribute boolean toClean.
+
+Step 2a. If the attribute toClean is equal to false, return the operation and give control back to the caller.
+
+Step 2b. If the attribute toClean is equal to true, access the storage data file and remove some of the activities starting from the oldest. Put these deleted activities into the data file under the ‘recycled’ folder.
+
+![image_info](./pictures/SDautoClean.png)
+
+#### 3.1.2 Additional Implementation (proposed)
+
+`StorageCleaner#setDeleteQuota - Allows the user to manipulate how much activity is deleted when auto cleanup is activated. Currently only delete the oldest 3 activities.`
+
+`StorageCleaner#setTimeFrame - Set a particular time frame for auto cleanup to activate. (i.e auto cleanup after every 2 weeks etc).`
+
+`StorageCleaner#initialiseLogCleaner - Gives the storage cleaner an added functionality of being able to clear log files too.`
+
+
+### 3.2 Storage feature
+The Storage class represents the back-end of Jikan, handling the creation, saving and loading of data. 
+Jikan uses a `.csv` file to store its data, formatted in the following way:
+
+`entry-name, start-time, end-time, duration, tags`
+
+All tags are saved in the same cell, separated by a white space; this design decision was taken to make sure that each entry occupies the same number of cells regardless of each entry’s number of tags. The tags are then separately parsed when the data is loaded.
+
+Each Storage objects contains the path to the data file (`Storage.dataFilePath`), the File object representing the data file (`Storage.dataFile`), and an activityList populated with the data from the data file (`Storage.activityList`). Storage optionally supports multiple data files at the same time, allowing implementation of features like multiple sessions and multiple user profiles. 
+
+Storage provides the following functions:
+- Constructing a Storage object via `Storage(String dataFilePath)`, which takes in the path to the desired data file (or the path where the user wants to create the data file) as a String object.
+- Creating a data file via `createDataFile`.
+- Writing to a data file via `writeToFile`. This function takes a single string as parameter and writes it to the data file. It is recommended to only pass single-line strings to keep the file nicely formatted. 
+Loading a pre-existing data file via `loadFile`. If a data file already exists for the provided data file path, the function will return `true`; if the specified data file did not previously exist, this function will call the `createDataFile` method and returns `false`. The return value is useful so that the application knows whether or not this is the first session with a specific data file or if data already exists.
```

Comment:
This portion is very lengthy, which makes it very confusing to understand. Perhaps you could break down this portion into smaller parts? 
You could also consider adding sequence diagrams showing the higher-level flow of the Storage class to help readers understand the flow of Storage class